### DISCUSSION TITLE
Support additional characters in routes

### DISCUSSION THREAD

**Discussion by AndrewBucklin** [postalserver/postal#1503] [Feature suggestions]
Route validation doesn't adhere to RFC. I noticed this when trying to add a route with an underscore in the email address. To fix it, I modified this file locally (line 37) and added the underscore:
https://github.com/postalhq/postal/blob/master/app/models/route.rb

But as per RFC, there are many other special characters allowed in the "local part" which this validation will incorrectly return as invaludl. Here's an article which goes into detail: https://help.returnpath.com/hc/en-us/articles/220560587-What-are-the-rules-for-email-address-syntax-

**Comment by willpower232** [comment]
Are you able to make a PR with the required changes in?

**Comment by plabbett** [comment]
Hello! I don't really know ruby well (or regex) but could you replace that line with something like this: `URI::MailTo::EMAIL_REGEXP`?

i.e.,

`validates :name, :presence => true, :format => /\A(([a-z0-9\-\.]*)|(\*)|(__returnpath__))\z/`

to

`validates :name, :presence => true, :format => URI::MailTo::EMAIL_REGEXP`

[Credit](https://stackoverflow.com/questions/22993545/ruby-email-validation-with-regex)

**Comment by plabbett** [comment]
I realize that's probably only the user section - these might be more appropriate

 `/\A([a-z0-9]+[\w|\-|\.|\+]*/i` according to [rubygarage.org](https://rubygarage.org/blog/how-to-validate-emails)

`/\A([\w+\-].?)/i` according to [emailregex.com](https://emailregex.com/)

**Comment by catphish** [comment]
This isn't a bug per-se. Just because special characters are allowed in email addresses doesn't mean a mailbox needs to support those characters. With that said, there's clearly a use case, and I see no reason not to support a reasonable variety of characters. I would suggest that `.` `_` and `-` would be sufficient, but if anyone thinks any other characters are likely to be useful, I will happily add them.

**Comment by AndrewBucklin** [comment]
I think + should also be supported since many people use the “plus addressing” feature I systems like Gmail and Office 365.

**Comment by willpower232** [comment]
seeing this has reminded me of https://github.com/postalhq/postal/issues/480#issuecomment-578123464 so there be similar use cases
